source/usage-examples/distinct.txt

```diff
@@ -0,0 +1,48 @@
+==========================================
+Distinct Values of a Field in a Collection
+==========================================
+
+.. default-domain:: mongodb
+
+You can retrieve a list of distinct values for a field across a collection
+by using the
+`distinct()
+<https://mongodb.github.io/node-mongodb-native/3.4/api/Collection.html#distinct>`_
+method. The ``distinct()`` method is called on a collection object and
+requires a document field name parameter. When called on a collection,
+``distinct()`` produces a list that contains one of each of the different
+values found in the specified document field.
+
+You can provide an optional query document to narrow the set of documents that
+are searched and an optional Object to specify additional query parameters.
+For details on the optional parameters, see the
+`distinct() method in the API documentation
+<https://mongodb.github.io/node-mongodb-native/3.4/api/Collection.html#distinct>`_.
+
+The ``distinct()`` method returns a
+`Promise <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise>`_
+that resolves to an
+`Array <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array>`_
+that contains each of the different values. If none of the documents in the
+collection contain the field specified in the method call, it returns an
+empty Array.
+
+If you specify a value for the document field name that is not a String such
```

Comment:
Not sure if we really need any more here, but what do you think about explaining this method's interaction with dot notation? (for instance, using field1.field2.field3 to query for values in nested objects?)
Other thought: arrays. What happens if I call this method on a field that is an array in some documents, but not others? Or a field that is an array in all documents in the collection? Does the type ``array`` show up, or individual values from arrays?

Reply:
Thanks for bringing up this point; I should include more detail to clarify these cases. The value supplied to the document field name parameter still needs to be a string. To specify a field subdocument, you use dot notation (e.g. "imdb.rating"). If you call it on an array (e.g. directors) field, distinct() treats each element as a value (rather than the array as a whole), but if you call it on a subdocument (e.g. imdb) field, the fields within the subdocument identify it as a unique value. E.g. 
```
[
  { rating: 6.6, votes: 8921, id: 86619 },
  { rating: 6.6, votes: 11781, id: 102713 },
  { rating: 6.4, votes: 10666, id: 117057 }
]
```